### PR TITLE
Usage Stats: Decouple from GrafanaLive

### PR DESCRIPTION
**Pull request by joanlopez**
**What this PR does / why we need it**:

The UsageStats service is a common source of cyclic dependencies due to its relationship with "third-party" services. 
So, the idea of this pull request is to invert the relationship between UsageStats service and GrafanaLive service to make the latter use the UsageStats service from a consumer perspective, as the other services does, instead of embedding it within the UsageStats service itself which is much more prone to cause cyclic dependencies.

**Special notes for your reviewer**:

The major tradeoff of this pull request was to either expose the update ticker through a method similar to `RegisterMetricsFunc` or to create a custom ticker for GrafanaLive service. So, I decided to go with the latter to avoid making the UsageStats service public API larger (which is something I'd like to avoid unless actually needed). Also, because that ticker it's not something that's going to change very often (so no big reasons to spread it / share it through an extensible mechanism) and this way we may have flexibility to set up a custom ticker length, if needed.

PS: Funny fact - the name of the branch is wrong, it's not about decoupling GrafanaLive from USS but the other way round! 😝 

Thanks! 🙌🏻

### CODE REVIEW

**Review by sakjur — APPROVED** [review]
:+1: LGTM, let's give the cloud team the chance to review as well

File: pkg/infra/usagestats/service/service.go [hunk 1]
```diff
@@ -110,12 +96,8 @@ func (uss *UsageStats) Run(ctx context.Context) error {
 				nextSendInterval = sendInterval
 				sendReportTicker.Reset(nextSendInterval)
 			}
-
-			// always reset live stats every report tick
-			uss.resetLiveStats()
```
Comment (FZambia) [inline pkg/infra/usagestats/service/service.go:115]: Noticed that we don't have live stats reset after sending in new code. Is this intentionally?

Comment (joanlopez) [reply pkg/infra/usagestats/service/service.go:115]: Hmm.. I'm not sure about that. 

I'd say the reset is still present but in a different place (look [here](https://github.com/grafana/grafana/blob/e0e1e9a6481df55b6eb9504fda5efa53aa0a8011/pkg/services/live/live.go#L1055)).

Does that make sense? 🤔  Thanks! 🙏🏻

File: pkg/services/live/live.go [hunk 2]
```diff
@@ -384,11 +387,31 @@ func (g *GrafanaLive) getStreamPlugin(pluginID string) (backend.StreamHandler, e
...
+
+	eGroup.Go(func() error {
+		updateStatsTicker := time.NewTicker(time.Minute * 30)
+		defer updateStatsTicker.Stop()
+
+		for {
+			select {
+			case <-updateStatsTicker.C:
+				g.sampleLiveStats()
+			case <-updateStatsTicker.C:
```
Comment (FZambia) [inline pkg/services/live/live.go:400]: Duplicate read from the same channel.

File: pkg/infra/usagestats/service/service.go (hunk 1 above)
Comment (FZambia) [reply pkg/infra/usagestats/service/service.go:115]: I see, thanks!

File: pkg/services/live/live.go (hunk 2 above)
Comment (joanlopez) [reply pkg/services/live/live.go:400]: Good catch! Copy-paste driven development is discouraged! 😛

Comment (joanlopez) [reply pkg/services/live/live.go]: Fixed, thanks!

**Review by undef1nd — APPROVED** [review]

**Review by FZambia — APPROVED** [review]
👍

**Comment by DanCech** [comment]
This changes the semantics of the grafana live stats reset, instead of being reset when a report is sent it's now reset when a report is generated.  This will cause problems when anything besides the uss service (eg licensing) calls the exported GetUsageReport function, because that will now reset the live stats.

**Comment by joanlopez** [comment]
> This changes the semantics of the grafana live stats reset, instead of being reset when a report is sent it's now reset when a report is generated. This will cause problems when anything besides the uss service (eg licensing) calls the exported GetUsageReport function, because that will now reset the live stats.

Does that mean that we need to let services register hook functions to be triggered once the report is sent? 🤔 

Any thought or alternative idea?

cc/ @sakjur @undef1nd

**Comment by DanCech** [comment]
I'm not sure what the best path forward is here, it may be to add support for a second callback that can be used to reset the stats as you suggest.

**Pull request merged**